File: java/src/main/be/steinsomers/bron_kerbosch/BronKerboschAlgorithm.java
package be.steinsomers.bron_kerbosch;

import java.util.function.Consumer;

@FunctionalInterface
public interface BronKerboschAlgorithm {
    int[] EMPTY_CLIQUE = {};

    void explore(UndirectedGraph graph, Consumer<int[]> cliqueConsumer) throws InterruptedException;
}

File: kotlin/src/main/be/steinsomers/bron_kerbosch/BronKerboschAlgorithm.kt
package be.steinsomers.bron_kerbosch

fun interface BronKerboschAlgorithm {
    @Throws(InterruptedException::class)
    fun explore(graph: UndirectedGraph, cliqueConsumer: CliqueConsumer)
}

data class CliqueConsumer(val minSize: Int, private val acceptor: (IntArray) -> Unit) {
    fun accept(clique: IntArray) {
        acceptor.invoke(clique)
    }
}
